ReCaptchaFrontendUi/view/frontend/web/js/nonInlineReCaptchaRenderer.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+define([
+    'jquery'
+], function ($) {
+    'use strict';
+
+    var reCaptchaEntities = [],
+        initialized = false,
+        rendererRecaptchaId = 'recaptcha-invisible',
+        rendererReCaptcha = null;
+
+    return {
+        add: function (reCaptchaEntity, parameters) {
+            if (parameters.size === 'invisible' && parameters.badge !== 'inline') {
+                if (!initialized) {
+                    this._init();
+                    grecaptcha.render(rendererRecaptchaId, parameters);
+                    setInterval(this._resolveVisibility, 100);
+                    initialized = true;
+                }
+
+                reCaptchaEntities.push(reCaptchaEntity);
+            }
+        },
+
+        _resolveVisibility: function () {
+            reCaptchaEntities.some(
+                (entity) => {
+                    return entity.is(":visible")
+                        // 900 is some magic z-index value of modal popups.
+                        && (entity.closest("[data-role='modal']").length == 0 || entity.zIndex() > 900)
```

Comment:
I not sure that find element with the closest is the best solution, we really haven't any options to find exactly needed element, instead of finding the closest? In this case, we depend on html layout that is not the best practice because we will not have the possibility to reuse this  component in another place where html layout will be different